### PR TITLE
Added remove_index

### PR DESCRIPTION
**Pull request by magec**
# WAT

Adds a check to disallow remove_index
# GIF

### CODE REVIEW

File: lib/rubocop/cop/migrations/remove_index.rb [hunk 1]
```diff
@@ -1,8 +1,31 @@
-module Rubocop
+module RuboCop
   module Cop
     module Migrations
-      # Your code goes here...
+      # Removing indexes is dangerous by nature, if the index you remove
+      # is used extensively, you could end up with database degradation.
```
Comment (andresgutgon) [inline lib/rubocop/cop/migrations/remove_index.rb:5]: That sounds to me :trollface:

File: spec/rubocop/cop/migrations/remove_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            end
+          end
+        MIGRATION
+      end
+
+      include_examples 'does not report any offenses'
+    end
+
+  end
+end
```
Comment (andresgutgon) [inline spec/rubocop/cop/migrations/remove_index_spec.rb:100]: What about testing when `rubocop:disable Migrations/RemoveIndex` comment is there?

File: lib/rubocop/cop/migrations/remove_index.rb (hunk 1 above)
Comment (magec) [reply lib/rubocop/cop/migrations/remove_index.rb:5]: hehe

File: spec/rubocop/cop/migrations/remove_index_spec.rb (hunk 2 above)
Comment (magec) [reply spec/rubocop/cop/migrations/remove_index_spec.rb:100]: That is part of rubocop code. I have testes it manually and it works, but that functionality is not related with a cop itself, but is about rubocop.

Comment (andresgutgon) [reply spec/rubocop/cop/migrations/remove_index_spec.rb:100]: got it

**Comment by andresgutgon** [comment]
👍

File: .rspec [hunk 3]
```diff
@@ -1,2 +1,3 @@
 --format documentation
 --color
+--require spec_helper
```
Comment (enricostano) [inline .rspec]: putting this here has bitten me tons of times, but probably someone else told you that it is awesome :trollface:

**Review by enricostano — CHANGES_REQUESTED** [review]

**Comment by enricostano** [comment]
:mailbox_with_no_mail: :+1:

File: lib/rubocop/cop/migrations/remove_index.rb (hunk 1 above)
Comment (sauloperez) [reply lib/rubocop/cop/migrations/remove_index.rb:5]: or simply downtime :trollface:

File: lib/rubocop/cop/migrations/remove_index.rb [hunk 4]
```diff
@@ -1,8 +1,29 @@
...
+      #
+      # This disallow the use of remove_index at all. It is only allowed
+      # inside a down method definition. If you actually know what you are doing
+      # (usually when the index is not being used) you should explicitly disable
+      # the check by adding a ``rubocop:disable Migrations/RemoveIndex`` comment
+      # in the offending line
       class RemoveIndex < Cop
+        MSG = 'remove_index is disalowed'.freeze
```
Comment (sauloperez) [inline lib/rubocop/cop/migrations/remove_index.rb]: s/disalowed/disallowed

File: rubocop-migrations.gemspec [hunk 5]
```diff
@@ -4,32 +4,25 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
...
+  spec.name          = 'rubocop-migrations'
   spec.version       = Rubocop::Migrations::VERSION
-  spec.authors       = ["enricostano"]
-  spec.email         = ["ocirne.onats@gmail.com"]
 
-  spec.summary       = ""
-  spec.description   = ""
-  spec.homepage      = ""
-  spec.license       = "MIT"
+  spec.authors       = %w(enricostano magec)
```
Comment (sauloperez) [inline rubocop-migrations.gemspec:10]: add your name as well

File: rubocop-migrations.gemspec [hunk 6]
```diff
@@ -4,32 +4,25 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
...
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
-  spec.require_paths = ["lib"]
+  spec.require_paths = ['lib']
 
   spec.add_runtime_dependency 'rubocop', '>= 0.42.0'
 
-  spec.add_development_dependency "bundler", "~> 1.11"
-  spec.add_development_dependency "rake", "~> 10.0"
-  spec.add_development_dependency "rspec", "3.5.0"
+  spec.add_development_dependency 'bundler', '~> 1.11'
```
Comment (sauloperez) [inline rubocop-migrations.gemspec:25]: I believe the latest is 1.13 something. I would relax it more to be able to use the latest.

File: .rspec (hunk 3 above)
Comment (sauloperez) [reply .rspec]: we don't use it on hosted and I'm not sure it's very common...

**Comment by franciscoj** [comment]
:clap: :+1:

File: .rspec (hunk 3 above)
Comment (magec) [reply .rspec]: I dropped it but does not show up... wonder why

Comment (magec) [reply .rspec]: Fixed

**Review by sauloperez — APPROVED** [review]

**Pull request merged**